### PR TITLE
ENH: parallel jobs of tfr across wavelets

### PR DESCRIPTION
**Pull request by kingjr**
This aims at optimizing the `_compute_tfr` function that I had cleaned up over the summer. It adds an option to parallelize across frequencies. It is computationally not quite as good when mkl is activated, but the memory load is much lower.

The usecase is intended towards audio decoding/STRF where we may need high time freq resolution of a few channels over long segments.

I know chris had his own implementation (couldn't find it anymore...), but I thought it'd be nice to have a single tfr function, rather than having too many redundancies.

### CODE REVIEW

File: mne/time_frequency/tfr.py [hunk 1]
```diff
@@ -356,26 +359,48 @@ def _compute_tfr(epoch_data, frequencies, sfreq=1.0, method='morlet',
...
+        # Parallelization is applied across frequencies.
+        parallel, my_cwt, n_jobs = parallel_func(_tfr_loop_freqs, n_jobs)
+        n_jobs = min(n_jobs, len(Ws[0]))
+        out = parallel(my_cwt(epoch_data, w, output, use_fft, decim, dtype)
+                       for w in np.array_split(Ws[0], n_jobs))
+        out = np.concatenate(out, axis=0)
+        if ('avg_' in output) or ('itc' in output):
+            out = out.transpose(1, 0, 2)  # from fct to cft
+        else:
+            out = out.transpose(2, 1, 0, 3)  # from cfet to ecft
```
Comment (jaeilepp) [inline mne/time_frequency/tfr.py]: Isn't this from cfet to efct?

Comment (kingjr) [inline mne/time_frequency/tfr.py]: you're right!

**Comment by Eric89GXL** [comment]
GitHub seems to be back for me. Go ahead and try to push again. Just have to wait out the DDoS if it fails again.

**Comment by agramfort** [comment]
can you give me numbers of speed gain on sample or your data?

**Comment by kingjr** [comment]
on my data it went from 53 s to 1 min 03 in terms of seep, but recruited
~.8 Go instead of 6Go

**Comment by agramfort** [comment]
can you share your benchmark script?

**Comment by kingjr** [comment]
```python
import numpy as np
from mne.time_frequency.tfr import _compute_tfr

X = np.random.randn(1, 1, 617412)
_compute_tfr(X, range(100, 400), sfreq=44100, decim=44, output='power',
             parallel_across='frequencies', n_jobs=-1)
_compute_tfr(X, range(100, 400), sfreq=44100, decim=44, output='power',
             parallel_across='channels', n_jobs=-1)
```

**Comment by agramfort** [comment]
this example uses only one channel.

the good solution would be to have parallel_across='auto'
but that requires a proper benchmark with M/EEG data.

**Comment by kingjr** [comment]
Feel free to take over, I'm only using for STRF with audio (hence 1-2
channels)

**Comment by kingjr** [comment]
I'm closing this for now. I use this implementation manually for now.

**Pull request closed without merging**